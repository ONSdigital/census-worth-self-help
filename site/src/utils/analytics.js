export default class Analytics {
  static trackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    } else {
      console.debug(`Track event not set : ${category} - ${action} - ${name} - ${value}`)
    }
  }

  static trackSiteSearch(keyword, category, searchCount) {
    if (window._paq) {
      window._paq.push(["trackSiteSearch", keyword, category, searchCount])
    } else {
      console.debug(`Track site search : ${keyword} - ${category} - ${searchCount}`)
    }
  }

  static setPageType(pageType) {
    if (typeof window !== `undefined` && window._paq) {
      window._paq.push(["setCustomDimension", 1, pageType])
    }
  }
}