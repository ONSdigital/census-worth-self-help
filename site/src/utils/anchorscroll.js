export default (history, timeout = 1000) => {
  let observer
  let timeoutId

  const reset = () => {
    if (timeoutId) {
      clearTimeout(timeoutId)
      timeoutId = null
    }

    if (observer) {
      observer.disconnect()
    }
  }

  const convertTextToLink = text => {
    return text.toLowerCase().replace(/\s/g, "-")
  }

  const createScrollToElement = hash => {
    return () => {
      let headers = []
      let tags = ["h1", "h2", "h3"]

      tags.forEach(tag => {
        headers = headers.concat(Array.from(document.getElementsByTagName(tag)))
      })

      let element = headers.find(
        header => convertTextToLink(header.textContent) === hash
      )

      if (element) {
        element.scrollIntoView()
        reset()
        return true
      }

      return false
    }
  }

  function scroll(hash) {
    if (typeof hash !== "string") {
      return
    }

    const anchorText = hash.slice(1).toLowerCase()

    if (!anchorText) {
      return
    }

    const scrollToElement = createScrollToElement(anchorText)

    setTimeout(() => {
      if (scrollToElement()) {
        return
      }

      if (window.MutationObserver) {
        observer = new MutationObserver(scrollToElement)

        observer.observe(document, {
          attributes: true,
          childList: true,
          subtree: true
        })
      }

      timeoutId = setTimeout(reset, timeout)
    })
  }

  // listen for changes
  history.listen(({ location, action }) => {
    if (timeoutId) {
      reset()
    }

    if (action !== "POP") {
      return
    }

    if (location.hash) {
      scroll(location.hash)
    }
  })

  // also trigger on load
  if (history.location.hash) {
    scroll(history.location.hash)
  }
}
