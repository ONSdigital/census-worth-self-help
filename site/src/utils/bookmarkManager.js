export default class BookmarkManager {
  constructor() {
    this.bookmarks = JSON.parse(localStorage.getItem("bookmarks"))
    if (this.bookmarks === null) {
      this.bookmarks = []
    }
  }

  getTopBookmarks() {
    return this.bookmarks
  }

  isPageBookmarked(page) {
    return this.bookmarks.includes(page)
  }

  bookmarkPage(page) {
    if (!this.isPageBookmarked(page)) {
      this.bookmarks.push(page)
    }
    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }

  unBookmarkPage(page) {
    this.bookmarks = this.bookmarks.filter(function(bookmark) {
      return bookmark !== page
    })

    localStorage.setItem("bookmarks", JSON.stringify(this.bookmarks))
  }
}
