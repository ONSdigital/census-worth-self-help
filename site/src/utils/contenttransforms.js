var sanitizeHtml = require("sanitize-html")

export const transformSources = htmlString => {
  let videoPath = process.env.GATSBY_ASSETS_PATH
    ? process.env.GATSBY_ASSETS_PATH
    : ""
  if (!htmlString) {
    return ""
  }
  return htmlString.replace(/{{TARGET_ASSETS_SRC}}/g, videoPath)
}

export const htmlSanitize = htmlString => {
  return sanitizeHtml(htmlString, {
    allowedTags: sanitizeHtml.defaults.allowedTags.filter(function(tag) {
      return tag !== "iframe"
    }).concat([
      "video",
      "audio",
      "img",
      "source",
      "h1",
      "h2"
    ]),
    allowedAttributes: {
      a: ["href", "name", "target"],
      img: ["src", "title"],
      source: ["src", "type"],
      video: ["controls"],
      audio: ["controls"]
    }
  })
}
