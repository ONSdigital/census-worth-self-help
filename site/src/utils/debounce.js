import _ from "lodash"

// debounce with default delay, or return sync function if delay exactly zero (for unit tests)
export default (fn, delay) => {
  return (delay === 0)
    ? fn
    : _.debounce((...args) => fn(...args), delay || 1000)
}