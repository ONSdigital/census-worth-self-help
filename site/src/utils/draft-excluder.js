const READY_FOR_PRODUCTION = "Ready for Live Site"

// Articles in draft should not be deployed to production. An article is draft if the draftreason for the article
// is anything other than "Ready for production" - see ONS-359.
//
// We also use this Draft Excluder concept to not show articles that are for the call centre only.
//
// Initially this was done using the draft tag. This technique might be removed in the future.
exports.excludeDraftArticle = node => {
  return Boolean(
    process.env.EXCLUDE_DRAFTS && (
      (
        node.frontmatter.tags &&
        node.frontmatter.tags.includes("draft")
      ) || (
        node.frontmatter.draftreason &&
        node.frontmatter.draftreason !== READY_FOR_PRODUCTION
      ) || node.frontmatter.cconly
    )
  )
}
