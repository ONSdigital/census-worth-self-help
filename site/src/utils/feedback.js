export default class Feedback {
  static bookmarkClickEvent(article) {
    Feedback.feedbackEvent("click-event", "bookmark", article)
  }
  static articleWasBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "bookmarked", article, 1)
  }
  static articleWasUnBookmarked(article) {
    Feedback.feedbackEvent("article-was-bookmarked", "unbookmarked", article, 1)
  }
  static articleIsUseful(article, reason) {
    Feedback.feedbackEvent(
        "article-feedback-review",
        "POSITIVE REVIEW: " + article,
        reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, 1)
  }
  static articleIsNotUseful(article, reason) {
    Feedback.feedbackEvent(
      "article-feedback-review",
      "REVIEW: " + article,
      reason
    )
    Feedback.feedbackEvent("article-feedback-rating", "rating", article, -1)
  }
  static feedbackEvent(category, action, name = "", value = "") {
    if (window._paq) {
      window._paq.push(["trackEvent", category, action, name, value])
    }
  }
}
