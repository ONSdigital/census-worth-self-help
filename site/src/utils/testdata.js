export const data = {
  edges: [
    {
      node: {
        frontmatter: {
          title: "test Article 1",
          description: "testDescription",
          priority: 1,
          date: new Date()
        },
        fields: {
          pagename: "test-article-1",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 2",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular", "webchat"]
        },
        fields: {
          pagename: "test-article-2",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 3",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-3",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 4",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["draft"]
        },
        fields: {
          pagename: "test-article-4",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 5",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-5",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Article 6",
          description: "testDescription",
          priority: 0,
          date: new Date(),
          tags: ["popular"]
        },
        fields: {
          pagename: "test-article-6",
          collection: "articles"
        }
      }
    },
    {
      node: {
        frontmatter: {
          title: "test Directory 7",
          description: "testDescription",
          priority: 0,
          date: new Date()
        },
        fields: {
          pagename: "test-directory-1",
          collection: "directories"
        }
      }
    }
  ]
}

export const articleNode = data.edges[0].node
export const webchatArticleNode = data.edges[1].node
export const directoryNode = data.edges[2].node
export const draftNode = data.edges[3].node
export const popularList = {
  edges: [data.edges[1], data.edges[2], data.edges[4], data.edges[5]]
}
export const articleList = { edges: data.edges.slice(0, 6) }
