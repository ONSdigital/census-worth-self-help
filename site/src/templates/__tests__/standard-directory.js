import React from "react"
import renderer from "react-test-renderer"

import StandardDirectory from "../standard-directory"

import {articleList} from "../../utils/testdata"

describe("StandardArticle", () => {
  const pageContext = {
  	breadcrumbs : [ {title: "root", link: "explore"} ],
    peers : [],
    description : "test description",
  	children : articleList.edges,
  	title : "test Directory 1"
  }

  it("renders correctly", () => {
    const tree = renderer
      .create(<StandardDirectory pageContext={pageContext} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})