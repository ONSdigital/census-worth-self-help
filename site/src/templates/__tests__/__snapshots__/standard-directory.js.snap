// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardArticle renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "vkixka",
      "next": undefined,
      "styles": "
        height: 100vh;
        display: flex;
        flex-direction: column;
      ",
    }
  }
>
  <header
    css={
      Object {
        "map": undefined,
        "name": "181ra4m",
        "next": undefined,
        "styles": "
        width: 100vw;
        
    background: linear-gradient(to left, rgb(0,61,89), #33637a);
  ;
      ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1thgah9",
          "next": undefined,
          "styles": "
            display: flex;
          ",
        }
      }
    >
      <div
        className="Header-Title-Style"
        css={
          Object {
            "map": undefined,
            "name": "17ih1bc",
            "next": undefined,
            "styles": "
              
    display: flex;
    justify-content: center;
    flex-direction: column;
  
              
    padding: 15px;
  
              flex-grow: 1;
            ",
          }
        }
        onClick={[Function]}
      >
        Self Help Facility
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "1dbjmka",
              "next": undefined,
              "styles": "
                  font-size: 19px;
                ",
            }
          }
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
      </div>
      <div
        className="Header-button-Style"
        css={
          Object {
            "map": undefined,
            "name": "1sm5jo",
            "next": undefined,
            "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        background-color: rgb(255, 255, 255, 0.2);
        border-radius: 4px;
        margin: 10px 10px 10px 0px;
        min-width: 20px;
        object-fit: contain;
        text-align: center;
        height: 40px;
      ",
          }
        }
        data-testid="topbarlink-button"
        onClick={[Function]}
      >
        Menu
      </div>
    </div>
  </header>
  <div
    css={
      Object {
        "map": undefined,
        "name": "o9ztju",
        "next": undefined,
        "styles": "
          
    margin: 0 auto;
    max-width: 700px;
  
          overflow: scroll;
          width: 100vw;
          display: flex;
          flex-direction: column;
          flex-grow: 1;
        ",
      }
    }
  >
    <div
      css={
        Object {
          "map": undefined,
          "name": "1k0aln6",
          "next": undefined,
          "styles": "
            flex-grow: 1;
          ",
        }
      }
    >
      <div
        css={
          Object {
            "map": undefined,
            "name": "ypl6yy",
            "next": undefined,
            "styles": "
        border-bottom: 1px rgb(229, 229, 234) solid;
        padding: 15px 0px;
      ",
          }
        }
      >
        <div
          className="Section-heading-Style"
          css={
            Object {
              "map": undefined,
              "name": "136nmlb",
              "next": undefined,
              "styles": "
          
    padding: 5px 15px;
  
        ",
            }
          }
        >
          EXPLORE CONTENT
        </div>
        <div
          css={
            Object {
              "map": undefined,
              "name": "136nmlb",
              "next": undefined,
              "styles": "
          
    padding: 5px 15px;
  
        ",
            }
          }
        >
          <div
            className="Button-subhead-Style css-2b097c-container"
            onKeyDown={[Function]}
          >
            <div
              className=" css-yk16xz-control"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <div
                className=" css-1hwfws3"
              >
                <div
                  className=" css-1wa3eu0-placeholder"
                >
                  Select...
                </div>
                <div
                  className="css-1g6gooi"
                >
                  <div
                    className=""
                    style={
                      Object {
                        "display": "inline-block",
                      }
                    }
                  >
                    <input
                      aria-autocomplete="list"
                      autoCapitalize="none"
                      autoComplete="off"
                      autoCorrect="off"
                      disabled={false}
                      id="react-select-2-input"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      spellCheck="false"
                      style={
                        Object {
                          "background": 0,
                          "border": 0,
                          "boxSizing": "content-box",
                          "color": "inherit",
                          "fontSize": "inherit",
                          "label": "input",
                          "opacity": 1,
                          "outline": 0,
                          "padding": 0,
                          "width": "1px",
                        }
                      }
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <div
                      style={
                        Object {
                          "height": 0,
                          "left": 0,
                          "overflow": "scroll",
                          "position": "absolute",
                          "top": 0,
                          "visibility": "hidden",
                          "whiteSpace": "pre",
                        }
                      }
                    >
                      
                    </div>
                  </div>
                </div>
              </div>
              <div
                className=" css-1wy0on6"
              >
                <div
                  aria-hidden="true"
                  className=" css-tlfecz-indicatorContainer"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <svg
                    aria-hidden="true"
                    className="css-19bqh2r"
                    focusable="false"
                    height={20}
                    viewBox="0 0 20 20"
                    width={20}
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "12ew49b",
            "next": undefined,
            "styles": "
        
    padding: 5px 15px;
  
      ",
          }
        }
      >
        <h1
          className="Article-Title-Style"
          css={
            Object {
              "map": undefined,
              "name": "1xcgh0k",
              "next": undefined,
              "styles": "
          margin-bottom: 0px;
          display: flex;
        ",
            }
          }
          data-testid="search-result-title"
        >
          test Directory 1
        </h1>
      </div>
      <div
        className="Article-body-Style"
        css={
          Object {
            "map": undefined,
            "name": "bsbyq4",
            "next": undefined,
            "styles": "
        
    padding: 5px 15px;
  ;
        margin-bottom: 20px;
      ",
          }
        }
      >
        <div
          className="Article-sub-title-Style"
        >
          test description
        </div>
      </div>
      <div
        css={
          Object {
            "map": undefined,
            "name": "1p7jpnp",
            "next": undefined,
            "styles": "
        
    padding: 15px 0px;
  ;
      ",
          }
        }
      >
        <div
          css={
            Object {
              "map": undefined,
              "name": "2artsq",
              "next": undefined,
              "styles": "
          
    padding: 15px 0px;
  ;
          
    padding: 5px 15px;
  
          display: flex;
          margin-bottom: 5px;
        ",
            }
          }
        >
          <div
            className="Tablist-heading-Style"
          >
            IN THIS SECTION:
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 1
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 2
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 3
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 4
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 5
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            css={
              Object {
                "map": undefined,
                "name": "lcxyxv",
                "next": undefined,
                "styles": "
        
    padding: 15px;
    min-height: 30px;
  ;
        box-shadow: 0px 2px 4px 0px rgba(0, 0, 0, 0.5);
        border-left: 6px solid rgb(0,61,89);
        background-color: rgb(255, 255, 255);
      ",
              }
            }
            onClick={[Function]}
          >
            <div
              className="Card-heading-Style"
              css={
                Object {
                  "map": undefined,
                  "name": "qiy98z",
                  "next": undefined,
                  "styles": "
          color: inherit;
          flex-grow: 1;
          font-weight: semi-bold;
        ",
                }
              }
              data-testid="articletab-article-card"
            >
              test Article 6
            </div>
            <div
              className="Card-sub-head-Style-gray"
              css={
                Object {
                  "map": undefined,
                  "name": "1hejrb1",
                  "next": undefined,
                  "styles": "
          padding-top: 6px;
          display: flex;
          flex-direction: vertical;
          line-height: 1.08;
          max-height: 42.12px; // line-height * font-size * 3
        ",
                }
              }
            >
              <div
                css={
                  Object {
                    "map": undefined,
                    "name": "lz6rtp",
                    "next": undefined,
                    "styles": "
            width: 70%;
            overflow: hidden;
            text-overflow: ellipsis;
            flex-grow: 1;
          ",
                  }
                }
              >
                testDescription
              </div>
              <div
                className="Card-meta-Style"
                css={
                  Object {
                    "map": undefined,
                    "name": "4j3tb8",
                    "next": undefined,
                    "styles": "
            display: flex;
            flex-direction: column;
            justify-content: flex-end;
          ",
                  }
                }
              >
                a few seconds ago
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      css={
        Object {
          "map": undefined,
          "name": "ej75ms",
          "next": undefined,
          "styles": "
        
    padding: 30px 0px;
  
        width: 100%;
        padding: 10px 0px;
        background-color: rgb(236, 236, 237);
        display: flex;
        min-height: 100px;
        flex-direction: column;
        justify-content: center;
        flex-shrink: 0;
      ",
        }
      }
    >
      <img
        alt="Footer logo"
        css={
          Object {
            "map": undefined,
            "name": "1v6eegl",
            "next": undefined,
            "styles": "
          margin-bottom: 50px;
          margin-top: 50px;
        ",
          }
        }
        src="/footer_logo.svg"
      />
      <div
        className="Footer-Style"
      >
        © Office for National Statistics 2019
      </div>
    </div>
  </div>
</div>
`;
